merlion/models/forecast/deep_base.py

```diff
@@ -0,0 +1,300 @@
+#
+# Copyright (c) 2022 salesforce.com, inc.
+# All rights reserved.
+# SPDX-License-Identifier: BSD-3-Clause
+# For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+#
+"""
+    Base class for Deep Learning Forecasting Models
+"""
+import copy
+import time
+
+import logging
+import numpy as np
+import pandas as pd
+
+
+from scipy.stats import norm
+from typing import List, Optional, Tuple, Union
+
+try:
+    import torch
+    import torch.nn as nn
+except ImportError as e:
+    err = (
+        "Try installing Merlion with optional dependencies using `pip install salesforce-merlion[deep-learning]` or "
+        "`pip install `salesforce-merlion[all]`"
+    )
+    raise ImportError(str(e) + ". " + err)
+
+
+from merlion.models.deep_base import DeepConfig, DeepModelBase
+from merlion.models.forecast.base import ForecasterBase, ForecasterConfig
+from merlion.models.utils.rolling_window_dataset import RollingWindowDataset
+from merlion.models.utils.time_features import get_time_features
+from merlion.models.utils.early_stopping import EarlyStopping
+
+from merlion.transform.base import TransformBase, Identity
+from merlion.transform.factory import TransformFactory
+from merlion.utils.misc import initializer, ProgressBar
+from merlion.utils.time_series import to_pd_datetime, to_timestamp, TimeSeries, AggregationPolicy, MissingValuePolicy
+
+
+logger = logging.getLogger(__name__)
+
+
+class DeepForecasterConfig(DeepConfig, ForecasterConfig):
+    def __init__(
+        self,
+        n_past: int,
+        start_token_len: int = 0,
+        **kwargs,
+    ):
+        super().__init__(
+            **kwargs,
+        )
+        self.n_past = n_past
+        self.start_token_len = start_token_len
+
+
+class DeepForecaster(DeepModelBase, ForecasterBase):
+    """
+    Base class for a deep forecaster model
+    """
+
+    config_class = DeepForecasterConfig
+
+    def __init__(self, config: DeepForecasterConfig):
+        super().__init__(config)
+
+    def _create_model(self):
+        self.deep_model = self.deep_model_class(self.config)
+
+        self._init_optimizer()
+
+        self._init_loss_fn()
+
+        self._post_model_creation()
+
+    def _init_optimizer(self):
+        self.optimizer = self.config.optimizer.value(
+            self.deep_model.parameters(),
+            lr=self.config.lr,
+            weight_decay=self.config.weight_decay,
+        )
+
+    def _init_loss_fn(self):
+        self.loss_fn = self.config.loss_fn.value()
+
+    def _post_model_creation(self):
+        logger.info("Finish model creation")
+
+    def _get_np_loss_and_prediction(self, evaluate_data: Union[pd.DataFrame, RollingWindowDataset]):
```

Comment:
Can you rename `evaluate_data` to `eval_dataset`? This simplifies the code below.